server/src/database/models/courseInstance.ts

```diff
@@ -0,0 +1,62 @@
+// SPDX-FileCopyrightText: 2022 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import { DataTypes, Model, InferAttributes, InferCreationAttributes, CreationOptional, ForeignKey } from 'sequelize';
+import { sequelize } from '..';
+import Course from './course';
+import User from './user';
+
+export default class CourseInstance extends Model<InferAttributes<CourseInstance>, InferCreationAttributes<CourseInstance>> {
+  declare id: CreationOptional<number>;
+  declare courseId: ForeignKey<Course['id']>;
+  declare gradingType: string;
+  declare startingPeriod: string;
+  declare endingPeriod: string;
+  declare teachingMethod: string;
+  declare responsibleTeacher: ForeignKey<User['id']>;
+  declare startDate: Date;
+  declare endDate: Date;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+}
+
+CourseInstance.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true
+    },
+    gradingType: {
+      type: DataTypes.ENUM('PASSFAIL', 'NUMERICAL'),
+      allowNull: false
+    },
+    startingPeriod: {
+      type: DataTypes.ENUM('I', 'II', 'III', 'IV', 'V'),
```

Comment:
Should we also consider the summer period for this? On the other hand, it probably could be easily added later.

Reply:
(in sisu) summer periods seems to be marked in plain text 'SUMMER' rather than with roman numerals. should we go with 'SUMMER' or 'VI' or change the enum to arabic numerals. Last option would make comparison operations easier.

Reply:
I will leave this as is, we can add the new value once we know the correct name for it.